lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2623,6 +2630,76 @@ submitExternalTx tr nw tl sealedTx = do
         pure sealedTx
     pure txid
 
+votingEraValidation
+    :: NetworkLayer IO block
+    -> ExceptT ErrConstructTx IO ()
+votingEraValidation nw = do
+    era <- liftIO $ currentNodeEra nw
+    case era of
+        Cardano.AnyCardanoEra Cardano.ConwayEra -> pure ()
+        _ -> throwE ErrConstructTxVotingInWrongEra
+
+checkingIfVoted
+    :: DBLayer IO s
+    -> NetworkLayer IO block
```

Comment:
```suggestion
    -> RecentEra era
```

You have an `era :: RecentEra era` very close-by at the call-sites! This way there's only one source of truth and no potential race conditions from reading the era twice.

Reply:
done